vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -269,6 +271,12 @@ def _get_min_max(targeted_data: pd.DataFrame, current_value=None) -> tuple[float
         _max = targeted_data.max(axis=None)
 
         # Convert to datetime if the column is datetime64
+        # AM question: I think this could be simplified:
+        #  * column type is already stored in _column_type and validated in __call__ with validate_column_type. Hence
+        #    we can just do if instance(self.selector, SELECTORS["temporal"]
+        #  * why do we need to do this type conversion at all? When we set min/max for datepicker in pre_build we don't
+        #  do any type
+        #    conversions.
```

Comment:
Great question, and this is how things are right now:
- `_min = targeted_data.min(axis=None)` -> sets `_min` to `datetime64` type (e.g: `2024-01-02T00:00:00.000000000`),
- `_min.item()` converts this to `integer` e.g. `1704067200000000000`
- This is still correct input for the `vm.DatePicker.min` as pydantic converts this integer to `datetime.date` e.g. `datetime.date(2024, 1, 1)`

By adding this explicit conversion of _min, _max, and current_value with pd.to_datetime, I ensured that the `_min.item()` does not convert it to integer (but it raises an exception that's suppressed). This is important because pydantic conversion from `int` to `datetime.date` will be bypassed when we support dynamic DatePicker as min and max will be directly propagated to the `dmc.DatePicker(...)` from the `vm.DatePicker.__call__()`. (I just realised that this could be solved by explicitly casting min and max from `vm.DatePicker.__call__()` to `date` type)

Also, this conversion ensures that the current_value is in the same format as _min and _max so we can calculate: `_min = min(_min, *current_value)`.

Reply:
This comment is irrelevant as the code is now much simpler and there's no more automatic conversion from  `datetime64` to `int`.